packages/dependency-extraction-webpack-plugin/lib/index.js

```diff
@@ -163,127 +163,104 @@ class DependencyExtractionWebpackPlugin {
+				entrypointChunks.add( chunk );
+			}
+		}
+
+		// Process each entrypoint chunk independently
+		for ( const chunk of entrypointChunks ) {
+			const chunkFiles = Array.from( chunk.files );
+
+			const chunkJSFile = chunkFiles.find( ( f ) => /\.js$/i.test( f ) );
+			if ( ! chunkJSFile ) {
+				// There's no JS file in this chunk, no work for us. Typically a `style.css` from cache group.
+				continue;
+			}
+
+			const chunkDeps = new Set();
 			if ( injectPolyfill ) {
-				entrypointExternalizedWpDeps.add( 'wp-polyfill' );
+				chunkDeps.add( 'wp-polyfill' );
 			}
 
 			const processModule = ( { userRequest } ) => {
 				if ( this.externalizedDeps.has( userRequest ) ) {
-					const scriptDependency = this.mapRequestToDependency(
-						userRequest
-					);
-					entrypointExternalizedWpDeps.add( scriptDependency );
+					chunkDeps.add( this.mapRequestToDependency( userRequest ) );
 				}
 			};
 
 			// Search for externalized modules in all chunks.
-			for ( const chunk of entrypoint.chunks ) {
-				const modulesIterable = isWebpack4
-					? chunk.modulesIterable
-					: compilation.chunkGraph.getChunkModules( chunk );
-				for ( const chunkModule of modulesIterable ) {
-					processModule( chunkModule );
-					// Loop through submodules of ConcatenatedModule.
-					if ( chunkModule.modules ) {
-						for ( const concatModule of chunkModule.modules ) {
-							processModule( concatModule );
-						}
+			const modulesIterable = isWebpack4
+				? chunk.modulesIterable
+				: compilation.chunkGraph.getChunkModules( chunk );
+			for ( const chunkModule of modulesIterable ) {
+				processModule( chunkModule );
+				// Loop through submodules of ConcatenatedModule.
+				if ( chunkModule.modules ) {
+					for ( const concatModule of chunkModule.modules ) {
+						processModule( concatModule );
 					}
 				}
 			}
 
+			// Go through the assets and hash the sources. We can't just use
+			// `chunk.contentHash` because that's not updated when
+			// assets are minified. In practice the hash is updated by
+			// `RealContentHashPlugin` after minification, but it only modifies
+			// already-produced asset filenames and the updated hash is not
+			// available to plugins.
 			const {
 				hashFunction,
 				hashDigest,
 				hashDigestLength,
 			} = compilation.outputOptions;
 
-			// Go through the assets and hash the sources. We can't just use
-			// `entrypointChunk.contentHash` because that's not updated when
-			// assets are minified. In practice the hash is updated by
-			// `RealContentHashPlugin` after minification, but it only modifies
-			// already-produced asset filenames and the updated hash is not
-			// available to plugins.
-			const hash = createHash( hashFunction );
-			for ( const filename of entrypoint.getFiles().sort() ) {
-				const asset = compilation.getAsset( filename );
-				hash.update( asset.source.buffer() );
-			}
-			const version = hash
+			const contentHash = chunkFiles
```

Comment:
I see you're dropping the `.sort()` here. Are we sure Webpack will always list the chunk's files in the same order?

It might be safer to keep sorting by filename to ensure a consistent ordering, just in case.

Reply:
I will add that change back before landing 👍🏻

Reply:
Done ✅